fabrics.c

```diff
@@ -299,13 +299,20 @@ static int __discover(nvme_ctrl_t c, const struct nvme_fabrics_config *defcfg,
 		else
 			print_discovery_log(log, numrec);
 	} else if (connect) {
+		int keep_alive_tmo = defcfg->keep_alive_tmo;
 		int i;
 
 		for (i = 0; i < numrec; i++) {
 			struct nvmf_disc_log_entry *e = &log->entries[i];
 			bool discover = false;
 			nvme_ctrl_t child;
 
+			if (e->subtype == NVME_NQN_DISC ||
+			    e->subtype == NVME_NQN_CURR)
+				set_discovery_kato(defcfg);
```

Comment:
Checking again, there still seems to be a problem here since set_discovery_kato() is still invoked in nvmf_discover() - so the I/O controllers will continue to get the wrong KATO values.

And we can't remove set_discovery_kato() from nvmf_discover() since the KATO values for the discovery controllers will no longer be set for the 'nvme discover' command. So this is a little tricky now...

Reply:
I think we just can drop the additional `set_discovery_kato()` in `nvmf_discover()`. `nvmf_discover()` calls eventually `__discover()` where we have the above code. Let me try to recreate this.

Reply:
Okay, got an idea how to address this. So we basically have to introduce setter function for the controller for these kind of attributes which are applied directly.

Reply:
Scratch this idea. We just need to make sure everytime we create a new discovery controller we just set the kato. This works well for all discovery paths, but not not for 'nvme connect'. Still trying to figure out what we could do there.